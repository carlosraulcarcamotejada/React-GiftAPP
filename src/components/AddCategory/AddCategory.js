import React, { useState } from 'react'
import PropTypes from 'prop-types'



const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleChangeOnInput = (e) => {
        setInputValue(e.target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setInputValue('');
            setCategories(categories => [inputValue,...categories]);
        }
    }

    return (
        <form onSubmit={onSubmit} >
            <input
                type="text"
                value={inputValue}
                onChange={handleChangeOnInput}
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};

export default AddCategory;

