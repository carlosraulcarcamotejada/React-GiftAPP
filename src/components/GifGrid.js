import React from 'react';
import useFetchGifs from '../CustomHooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    

    return (
        <>
            <h3> {category}  </h3>
            { loading && <p >Loading...</p>}
            {
                images.map(image =>
                    <GifItem
                        key={image.id}
                        {...image}
                    />)
            }

        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}