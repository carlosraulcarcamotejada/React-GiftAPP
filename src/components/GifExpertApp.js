import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'




export const GifExpertApp = ({ categoriesDefault = [] }) => {

    const [categories, setCategories] = useState(categoriesDefault)



    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
            />
            <hr />

            {
                categories.map(category => {
                    return (<GifGrid key={category} category={category} />);
                })
            }

        </>
    )
}

