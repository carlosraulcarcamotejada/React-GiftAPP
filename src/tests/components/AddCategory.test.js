import { shallow } from 'enzyme';
import React from 'react';
import { AddCategory } from '../../components/AddCategory'
import '@testing-library/jest-dom';


describe('Test en el componenten <AddCategory />', () => {


    const setCategories = jest.fn();

    let wrapper = shallow(<AddCategory setCategories={setCategories} />);


    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });




    test('Test 1. Debe mostrar <AddCategory setCategories={setCategories} /> correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });


    test('Test 2. Debe cambiar el valor del input', () => {


        const value = 'Hola Mundo';

        wrapper.find('input').simulate('change', { target: { value } });

        expect(wrapper.find('input').prop('value')).toBe(value);
    });




    test('Test 3 no debe postear la informaciÃ³n  ', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } })

        expect(setCategories).not.toHaveBeenCalled();

    })


    test('Test 4 debe llamar setCategories y limpiar la caja', () => {

        const value = 'Hello World'

        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).toHaveBeenCalled();

        expect(wrapper.find('input').prop('value')).toBe('');


    })


});

