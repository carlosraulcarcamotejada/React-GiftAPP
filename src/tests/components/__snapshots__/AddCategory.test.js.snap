// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en el componenten <AddCategory /> Test 1. Debe mostrar <AddCategory setCategories={setCategories} /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
